src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisResultService.java

```diff
@@ -167,8 +172,23 @@ public void insertStatus(List<UUID> resultUuids, SecurityAnalysisStatus status)
 
     @Transactional
     public void delete(UUID resultUuid) {
+        AtomicReference<Long> startTime = new AtomicReference<>();
+        startTime.set(System.nanoTime());
         Objects.requireNonNull(resultUuid);
-        securityAnalysisResultRepository.deleteById(resultUuid);
+        deleteWithSQLQueries(resultUuid);
+        LOGGER.info("Security analysis result '{}' has been deleted in {}ms", resultUuid, TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime.get()));
+    }
+
+    // We manually delete the results here using SQL queries to improve performances.
+    // source : https://www.baeldung.com/spring-data-jpa-deleteby
+    // "The @Query method creates a single SQL query against the database. By comparison, the deleteBy methods execute a read query, then delete each of the items one by one."
+    private void deleteWithSQLQueries(UUID resultId) {
```

Comment:
I'm not sure it's a good namming "...WithSQLQueries" maybe just do like in the SC PR -> `deleteSecurityAnalysisResult`

Reply:
DOne